EXTRA_FIM/pyiron_job.py

```diff
@@ -0,0 +1,247 @@
+from pathlib import Path
+from pyiron_base.utils.error import ImportAlarm
+from pyiron_base.jobs.master.parallel import ParallelMaster
+from pyiron_base.jobs.job.jobtype import JobType
+from pyiron_base import JobGenerator
+from pyiron_base.jobs.job.template import TemplateJob
+import h5py
+import numpy as np
+from pyiron_atomistics.sphinx.structure import read_atoms
+
+try:
+    import EXTRA_FIM.main as fim
+    from EXTRA_FIM.potential import extend_potential, sx_el_potential3D_cell
+    from EXTRA_FIM.datautils.pre_processing import suggest_input_dictionary
+    from EXTRA_FIM.sx_nc_waves_reader import sx_nc_waves_reader
+except ImportError:
+    import_alarm = ImportAlarm("Unable to import EXTRA_FIM")
+
+__author__ = "Christoph Freysoldt, Shyam Katnagallu"
+__copyright__ = (
+    "Copyright 2021, Max-Planck-Institut für Eisenforschung GmbH - "
+    "Computational Materials Design (CM) Department"
+)
+__version__ = "0.1.0"
+__maintainer__ = "Shyam Katnagallu"
+__email__ = "s.katnagallu@mpie.de"
+__status__ = "development"
+__date__ = "March 5, 2024"
+
+
+class ExtraFimSimulatorRefJob(TemplateJob):
+    """Reference pyiron job for Extra FIM simulation."""
+
+    def __init__(self, project, job_name):
+        super().__init__(project, job_name)
+        self.input["waves_directory"] = None
+        self.input["waves_reader"] = None
+        self.input["kpoint"] = None
+        self.input["ionization_energies"] = None
+        self.input["extrapolate_potential"] = False
+        self.input["total_kpoints"] = None
+
+    def extrpolate_potential(self):
+        """Extrapolate potential if needed, to extrapolate waves to higher distances"""
+
+        elec_potential, _ = sx_el_potential3D_cell(
+            self.input["simulator_dict"]["working_directory"]
+        )
+        pot, _, _, cell = fim.potential(self.input["simulator_dict"]).potential_cell()
+
+        if self.input.extrapolate_potential:
```

Comment:
I am confused about this if.

Is this function supposed to be called by the user before running, or as part of the running job (internal function)?

If this is not meant to be used by the user (who would rather set the input variables), the function should probably begin with an underscore to mark it as internal/private function. If this is to be run by user, where is the input dict set up to actually do something?